test/manifest.ts

```diff
@@ -14,22 +14,560 @@
 
 import {describe, it, afterEach} from 'mocha';
 
-import {Manifest} from '../src/manifest';
+import {Manifest, ManifestConfig} from '../src/manifest';
 import * as sinon from 'sinon';
-import {GitHub} from '../src/github';
-import {stubFilesFromFixtures} from './releasers/utils';
+import {GitHub, MergedGitHubPRWithFiles} from '../src/github';
+import {stubFilesFromFixtures, buildGitHubFileRaw} from './releasers/utils';
 import {expect} from 'chai';
+import {Commit} from '../src/graphql-to-commits';
 import {CheckpointType} from '../src/util/checkpoint';
+import {stubSuggesterWithSnapshot, buildMockCommit} from './helpers';
 
 const fixturePath = './test/fixtures/manifest/repo';
 const defaultBranch = 'main';
+const lastReleaseSha = 'abc123';
 const sandbox = sinon.createSandbox();
 
 describe('Manifest', () => {
   afterEach(() => {
     sandbox.restore();
   });
 
+  function stubGithub(options: {
+    github: GitHub;
+    commits?: Commit[];
+    manifest?: string;
+    lastReleaseSha?: string;
+    mergedPRFiles?: string[];
+    addLabel?: string | false;
+  }) {
+    const {
+      github,
+      commits,
+      manifest,
+      lastReleaseSha,
+      mergedPRFiles,
+      addLabel,
+    } = options;
+    if (manifest !== undefined && lastReleaseSha) {
+      sandbox
+        .stub(github, 'getFileContentsWithSimpleAPI')
+        .withArgs('.release-please-manifest.json', lastReleaseSha, false)
+        .resolves(buildGitHubFileRaw(manifest));
+    }
+    let mergedPR: MergedGitHubPRWithFiles | undefined;
+    if (lastReleaseSha) {
+      mergedPR = {
+        sha: lastReleaseSha,
+        title: '',
+        body: '',
+        number: 22,
+        baseRefName: '',
+        headRefName: '',
+        files: mergedPRFiles ?? [],
+        labels: [],
+      };
+    }
+    sandbox.stub(github, 'lastMergedPRByHeadBranch').resolves(mergedPR);
+    sandbox
+      .stub(github, 'findFilesByFilename')
+      // implementation strips leading `path`
+      .resolves(['src/foolib/version.py']);
+    // creating a new PR
+    sandbox.stub(github, 'findOpenReleasePRs').resolves([]);
+    sandbox
+      .stub(github, 'commitsSinceSha')
+      .withArgs(lastReleaseSha)
+      .resolves(commits ?? []);
+
+    const addLabelsStub = sandbox.stub(github, 'addLabels');
+    const labelToAdd = addLabel ?? 'autorelease: pending';
+    if (labelToAdd) {
+      addLabelsStub.withArgs([labelToAdd], 22).resolves(true);
+    }
+    addLabelsStub.throws(() => {
+      const argsArr = addLabelsStub.getCalls().map(c => c.args);
+      const unexpected = argsArr.filter(args => args[0][0] !== labelToAdd);
+      return new Error(
+        'addLabelsStub called with unexpected args: ' +
+          JSON.stringify(unexpected[0])
+      );
+    });
+  }
```

Comment:
I was getting really frustrated with a stub silently producing undefined if the `withArgs` don't match - I want to know when it's called unexpectedly! This the best I could figure out how to do from the docs